### PR TITLE
Dependencies updated

### PR DESCRIPTION
**Pull request by RijnBuve-TomTom**
Neo code changes, just `pom.xml` updates.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -95,25 +95,26 @@
...
         <!-- Library versions. -->
-        <equalsverifier.version>3.14</equalsverifier.version>
+        <equalsverifier.version>3.15.2</equalsverifier.version>
         <junit.version>4.13.2</junit.version>
-        <junit-jupiter.version>5.9.2</junit-jupiter.version>
-        <kotlin.version>1.8.10</kotlin.version>
-        <kotlin-test.version>1.8.10</kotlin-test.version>
-        <kotlin-test-junit.version>1.8.10</kotlin-test-junit.version>
+        <junit-jupiter.version>5.10.0</junit-jupiter.version>
+        <kotlin.version>1.9.20-RC2</kotlin.version>
```
Comment (JeroenErikJensen-TomTom) [inline pom.xml]: This forces all clients to use this release candidate in production. Let's just use the latest release.

```suggestion
        <kotlin.version>1.9.10</kotlin.version>
```

same below

**Review by JeroenErikJensen-TomTom — CHANGES_REQUESTED** [review]

**Review by OresteSalerno-TomTom — DISMISSED** [review]

**Review by JeroenErikJensen-TomTom — DISMISSED** [review]

**Comment by OresteSalerno-TomTom** [comment]
It seems that the TODOs I added in `IfExtensionsTest` as part of version 1.8.3 are not necessary anymore, even though [this issue](https://github.com/mockk/mockk/issues/1033) is still open. So maybe you can also address those TODOs as part of this release.

File: extensions/src/test/java/com/tomtom/kotlin/extensions/IfExtensionsTest.kt [hunk 2]
```diff
@@ -100,7 +100,6 @@ internal class IfExtensionsTest {
     fun `ifNull on null`() {
         // GIVEN
         val sut: Int? = null
-        // TODO: replace with `spyk({ 1 })` when https://github.com/mockk/mockk/issues/1033 is fixed.
         val ifNullBlock: () -> Int = spyk {
             every { this@spyk.invoke() } answers { 1 }
         }
```
Comment (OresteSalerno-TomTom) [inline extensions/src/test/java/com/tomtom/kotlin/extensions/IfExtensionsTest.kt]: I meant that you can *resolve* the TODOs, not just remove them.
For example, this block can be changed into (as it used to be):
```suggestion
        val ifNullBlock = spyk({ 1 })
```

Also, there are more TODOs in this file than the ones you removed.

**Review by OresteSalerno-TomTom — CHANGES_REQUESTED** [review]

**Review by JeroenErikJensen-TomTom — DISMISSED** [review]

File: extensions/src/test/java/com/tomtom/kotlin/extensions/IfExtensionsTest.kt (hunk 2 above)
Comment (RijnBuve-TomTom) [reply extensions/src/test/java/com/tomtom/kotlin/extensions/IfExtensionsTest.kt]: Resolved

Comment (OresteSalerno-TomTom) [reply extensions/src/test/java/com/tomtom/kotlin/extensions/IfExtensionsTest.kt]: The other TODOs were still not addressed, I figured it's faster to push a commit with what I was requesting. :)

Comment (RijnBuve-TomTom) [reply extensions/src/test/java/com/tomtom/kotlin/extensions/IfExtensionsTest.kt]: Thanks

**Review by OresteSalerno-TomTom — APPROVED** [review]

**Pull request merged**